psp/scripts/eval_model.py

```diff
@@ -86,6 +86,13 @@
     help="The time interval of the samples in minutes. "
     "Defaults to None and then value is taken from the configuration.",
 )
+@click.option(
```

Comment:
BTW passing an evaluation file through the `--eval-config` is already possible (although, granted, this is not the clearest part of the codebase!).

Also instead of adding a "fork" in `PvXDataPipe`, it would be more consistent to make a subclass of `PvDataSource` that supports your CSV format (`CsvPvDataSource`), just like we have `NetcdfPvDataSource` for pv data in `.netcdf` files.

Reply:
but the CSV is just for getting X (pv_id and ts), the netcdf is still needed. 

I could imagine make a `PVXDataPipeFromCsv` that loaded it from the csv. That might be neater

Reply:
Ok yes I see now. You are right the neat way would be a subclass of `IterDataPipe[X]` that generates `(pv_id, ts)` samples from a csv file. The `make_data_loader` function should probably be made to **receive** such a `IterDataPipe[X]` instead of creating the right one given the arguments (this would clean up the `if else` that's currently deciding between the `PvXDataPipe` and `RandomPvXDataPipe`).

Reply:
I dont mind `make_data_loader` choosing the right one. Wraps it all up into a `thing` that make the correct `data_loader`

Reply:
But you'll end up with something more and more complex as you want to support more ways to generate your `(pv_id, ts)` samples:


```python
def make_data_loader(
  ... some args for the PvDataSource version,
  shuffle: bool,
  ... some args for the csv version,
  # some arguments that are independent of the datapipe:
  batch_size: int,
  num_workers: int,
):
...
if csv_file is not None:
    datapipe = PvXDataPipe(...args from the pv data source )
elif shuffle:
   datapipe = RandomPvXDataPipe(...args from the pv data source)
else:
   datapipe = CsvPvXDataPipe(...args for the csv version)

```

Injecting it would be cleaner:

```python
def make_data_loader(
  datapipe: DataIterPipe[X],
  batch_size: int,
  num_workers: int,
):
```

No "merging of all the possible arguments for all the possible `DataIterPipe[X]` in the function. The drawback will be adapting the config so that you can specify which `DataIterPipe[X]` to use in there, which shouldn't be too tricky in theory.